controllers/metal3.io/host_state_machine.go

```diff
@@ -428,7 +428,26 @@ func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
 }
 
 func (hsm *hostStateMachine) provisioningCancelled() bool {
-	return hsm.imageProvisioningCancelled() && hsm.customDeployProvisioningCancelled()
+	if hsm.Host.Spec.CustomDeploy != nil {
+		if hsm.Host.Spec.CustomDeploy.Method == "" {
+			return true
+		}
+		if hsm.Host.Status.Provisioning.CustomDeploy != nil && hsm.Host.Spec.CustomDeploy.Method != hsm.Host.Status.Provisioning.CustomDeploy.Method {
+			return true
+		}
+		// At this point spec.CustomDeploy value didn't change (and it's not empty). Only a discrepancy in the Image
+		// could require a deprovisioning, but if it isn't set or empty then is fine
+		if hsm.Host.Spec.Image != nil && hsm.Host.Status.Provisioning.Image.URL != "" && hsm.Host.Spec.Image.URL != hsm.Host.Status.Provisioning.Image.URL {
+			return true
+		}
+
+		return false
+	} else if hsm.Host.Status.Provisioning.CustomDeploy != nil && hsm.Host.Status.Provisioning.CustomDeploy.Method != "" {
+		return true
+	}
+
+	// Fallback to the previous case
```

Comment:
nit: 'previous' here refers to history not to control flow, so it will make less sense outside the context of the code review.